### ISSUE TITLE
doc.schema.childSchemas and doc.schema.paths[foo].schema

### ISSUE THREAD

**Issue by Spown** [Automattic/mongoose#5695]
I don't know when exactly, but somewhere between 4.10 and 4.12 ``schema.childSchemas[foo]`` and ``schema.paths[bar].schema`` from a document, that were pointing to the same schema stopped being references to the same objects. The two still describe the same schemas (have the same paths, options, itc), but they appear to be constructed independently.

```js
		for (var schemaPath in schema.paths) {
			if (schema.paths.hasOwnProperty(schemaPath)) {
				var schemaField = schema.paths[schemaPath];
				if (schema.childSchemas.indexOf(schemaField.schema) !== -1) {
```
 In the code snippet above I perform a search/check for a nested field constructed from a child schema. In <=4.10 this would be true if the field was indeed of a child schema. Now it is always ``false``. 

Thank you.

**Comment by Spown** [comment]
Please tell me if this is the intended behavior or a bug so that I could either focus on a workaround for [my plugin fork](/Spown/mongoose-i18n-localize/blob/fe5b40789868a1e67b21754de01a807ea5f87150/index.js#L57) or wait for a fix.

**Comment by varunjayaraman** [comment]
yeah, so it looks like this is definitely now no longer the case:

```javascript
const mongoose = require('mongoose');
const co = require('co');
const assert = require('assert');
mongoose.Promise = global.Promise;
const GITHUB_ISSUE = `gh-5695`


exec()
  .then(() => {
    console.log('successfully ran program');
    process.exit(0);
  })
  .catch(error => {
    console.error(`Error: ${ error }\n${ error.stack }`);
  });


function exec() {
  return co(function*() {
    const childSchema = new mongoose.Schema({
      age: Number
    });
    
    const schema = new mongoose.Schema({
      name: String,
      child: childSchema
    });

    assert.ok(
      Object.is(schema.paths['child'].schema, schema.childSchemas[0])); // fails on 4.12, passes on 4.9
  });
}
```

**Comment by vkarpov15** [comment]
I bisected this down to 4.11.4 because of [this change](https://github.com/Automattic/mongoose/compare/2b337ddcf6ac2b8357a4c23a1926f0a448826843...ee1987eab5b5d47bc7bb2e6736d77d506970a92d#diff-f412f4722a52cf22ed3d0a9a694b837d) . The below assert works in `>= 4.11.4`

```javascript
    assert.ok(
      Object.is(schema.paths['child'].schema, schema.childSchemas[0].schema)); // succeeds on >= 4.11.4
```

`childSchemas` now contains an array of objects that have a `schema` property and a `model` property. Sorry about the inconvenience but `childSchemas` wasn't formally part of the public-facing API so we changed its structure to enable us to fix #5437 (see af73f649cdcfe00bc2d2cea64a68bed8b1f7c943). Since you're depending on it we'll add tests to make sure the structure doesn't change underneath you anymore :+1:
